rust/main/submitter/src/chain_tx_adapter/transaction.rs

```diff
@@ -0,0 +1,33 @@
+// TODO: re-enable clippy warnings
+#![allow(dead_code)]
+
+use hyperlane_core::H256;
+use uuid::Uuid;
+
+type PayloadId = Uuid;
+type SignerAddress = H256;
+
+/// Full details about a transaction
+pub struct Transaction {
+    uuid: Uuid,                         // unique tx identifier. Used as primary key in the db.
+    tx_hash: Option<H256>, // tx identifier obtained by hashing its contents. This may change when gas price is escalated
```

Comment:
nit: rename field `tx_hash` into `hash` since it is inside of `Transaction` struct.

```suggestion
    hash: Option<H256>, // tx identifier obtained by hashing its contents. This may change when gas price is escalated
```

And I wonder if we want to introduce an alias for `H256` when it is used for a hash (as contrast against `SignerAddress`).

Reply:
will rename the field, but I think H256 stands for "hash" so I think it's descriptive enough

Reply:
I don't mind using `H256`. Last time we needed to migrate from `H256` to `H512`, we changed it in many places.

By the way, we are using `H512` in `HyperlaneProvider` as transaction hash type.

Reply:
happy to use an alias everywhere but it feels like scope creep for this particular PR

Reply:
great point on H512, will use that